rust/src/lightning.rs

```diff
@@ -525,6 +528,110 @@ pub async fn run_ldk(system: &LightningSystem) -> Result<BackgroundProcessor> {
     Ok(background_processor)
 }
 
+pub async fn run_ldk_server(
```

Comment:
I suspect we could just use `run_ldk` on both sides (and just listen for incoming connections on both). Two different functions, doing almost the same, make things more confusing than actually solve.

Reply:
in the meantime, to make things more maintainable, I just refactored the code a bit (so it calls common path afterwards) - this should make it much less confusing.